src/ImageSharp.Web/DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -121,7 +95,19 @@ public static IImageSharpBuilder AddImageSharp(this IServiceCollection services)
             builder.AddConverter<ColorConverter>();
             builder.AddConverter<EnumConverter>();
 
-            builder.Services.AddSingleton<CommandParser>();
+            return builder;
+        }
+
+        private static IImageSharpBuilder AddDefaultServices(this IImageSharpBuilder builder)
```

Comment:
Although default implementations are also set for `IRequestParser`, `IImageCache`, `ICacheKey` and `ICacheHash`, those are actually required for `ImageSharpMiddleware` to activate/resolve from the service container and can easily be replaced with the respective set-methods.

Maybe this extension method should be made public and the `AddImageSharp()` shouldn't call it by default, so you can opt-in to adding these default providers/processors? And following this pattern, maybe even split it into `AddDefaultProviders()` and `AddDefaultProcessors()`?

Reply:
This would also solve the issue I described in PR https://github.com/SixLabors/ImageSharp.Web/pull/185: the order of registering `IImageProvider`s is important, but can't be (easily) changed. Having a separate method to add the default provider after your own will remove the need to remove and re-add them...

Reply:
I don’t want to change registration methods at this time.

Reply:
The whole point of the default implementation is that out of the box it works.

Reply:
I wasn't sure about this either and agree that it's better to have everything working OOTB 😄 I see you've also made inserting providers easier in https://github.com/SixLabors/ImageSharp.Web/pull/247, so this is now less of an issue...